### PR TITLE
Add gloss coverage reporting, optional fingerspelling fallback, and iterative lookup normalization

### PR DESCRIPTION
**Pull request by GerrySant**
This PR introduces several improvements to the gloss-to-pose lookup pipeline, aimed at increasing robustness, transparency, and user control.

### New features
- **Gloss coverage reporting** (`--coverage-info`): enables computation and reporting of the ratio of glosses successfully mapped to poses during lookup.
- **Optional fingerspelling fallback** (`--no-use-fingerspelling`): allows disabling fingerspelling as a backup mechanism when a gloss is not found in the lexicon, giving users explicit control over lookup behavior.

### Iterative lookup with progressive normalization
- The gloss lookup strategy has been extended with an iterative, rule-based normalization pipeline.
- When a gloss is not found in the lexicon, multiple lookup attempts are performed using progressively more aggressive transformations, including:
  - lowercasing and stripping,
  - rule-based cleanup (e.g. removing '+' and trailing '-ix'),
  - SpaCy-lemma–style cleanup,
  - normalization of standalone integer tokens with surrounding punctuation,
  - filtering to keep only Unicode letters.
- Each transformation is applied cumulatively, and lookup is retried after each step.
- This approach significantly improves matching robustness while preserving the original gloss whenever possible.

### Internal improvements
- Fixed a thread-safety issue when loading poses in parallel. `Pose.read()` is not thread-safe and was previously invoked concurrently from multiple threads, leading to race conditions in the internal binary reader and intermittent NumPy errors such as *"buffer is too small for requested array"*.
  - Pose files are now read in parallel, but decoding via `Pose.read()` is serialized using a global lock to ensure thread safety.
- Updated `lookup_sequence()` to force eager evaluation of `ThreadPoolExecutor` results by materializing `executor.map()` into a list.
  - This guarantees all lookup threads complete before further processing.
  - Enables safe computation of gloss coverage statistics.
  - Avoids subtle bugs related to lazy iterator reuse.

These changes improve lookup success rates, robustness under parallel execution, and observability of gloss-to-pose coverage, without altering default behavior unless explicitly requested by the user.

### CODE REVIEW

File: spoken_to_signed/gloss_to_pose/lookup/fingerspelling_lookup.py [hunk 1]
```diff
@@ -36,7 +36,9 @@ def characters_lookup(self, word: str, spoken_language: str, signed_language: st
                     break
 
             if not found:
-                raise FileNotFoundError(f"Characters {word} not found in fingerspelling lexicon")
+                raise FileNotFoundError(f"[WARN] Characters {word} not found in fingerspelling lexicon")
+            else:
+                print(f"[SUCCESS] fingerspelling for: {word} FOUND!")
```
Comment (bricksdont) [inline spoken_to_signed/gloss_to_pose/lookup/fingerspelling_lookup.py]: I would remove this print statement for when operations succeed / everything is as expected

File: spoken_to_signed/gloss_to_pose/lookup/lookup.py [hunk 2]
```diff
@@ -84,49 +189,117 @@ def get_best_row(self, rows, term: str):
...
+                    preview_gloss = next_step_fn(current_gloss)
+                    next_msg = f" Trying next gloss version: '{preview_gloss}'."
+                else:
+                    # The next step is the "no-change" attempt (just advance the index)
+                    next_msg = " Trying next lookup configuration."
+            else:
+                # No more local steps, will move to backups
+                next_msg = " No more preprocessing steps. Will try backup strategies (if available)."
+
+            print(f"[GLOSS '{original_gloss}'] [FAIL #{step_idx}] Lookup failed for gloss='{current_gloss}'.{next_msg}")
```
Comment (bricksdont) [inline spoken_to_signed/gloss_to_pose/lookup/lookup.py]: This printing / reporting is quite verbose in my opinion, is it really necessary for normal operation?

**Review by AmitMY — CHANGES_REQUESTED** [review]

**Comment by AmitMY** [comment]
Closing the umbrella PR: all of its topics have now landed or are landing as focused PRs — coverage reporting in #78, progressive lookup normalization in #81, and disabling fingerspelling landed earlier as #64 (docs in #80). The thread-safety issue around `Pose.read` is acknowledged as real (it reproduces on unmodified `main`) and should be fixed in pose-format. Thanks @GerrySant for splitting this up and iterating.
